net/gensync/Portfile

```diff
@@ -1,10 +1,17 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
 PortSystem              1.0
+
+name                    gensync
 PortGroup               cmake 1.1
 PortGroup               github 1.0
+PortGroup               legacysupport 1.1
+
+# strndup
+legacysupport.newest_darwin_requires_legacy 10
+compiler.cxx_standard   2011
 
-github.setup            nislab gensync 1.0.4 v
+github.setup            nislab gensync-macports 2.0.4 v
```

Comment:
@ryandesign Thanks for reviewing our request! The reason for the separate repository is that the gensync-macports repository contains a different CMakeLists.txt than our main gensync repository. We did this on purpose because we do not need some of the functionality of the original CMakeLists.

Reply:
Just to be clear ... we have other repos that use the same gensync core code but have specialized utility (e.g., benchmarking).

Reply:
> @ryandesign Thanks for reviewing our request! The reason for the separate repository is that the gensync-macports repository contains a different CMakeLists.txt than our main gensync repository. We did this on purpose because we do not need some of the functionality of the original CMakeLists.

@nathanstrahs If changes are minor, you could make a patch for CMakeLists. If changes are huge, then make `CMakeLists.txt.macports` file and add something like this to the portfile:
```
post-extract {
    delete ${worksrcpath}/CMakeLists.txt
    move ${worksrcpath}/CMakeLists.txt.macports ${worksrcpath}/CMakeLists.txt
}
And use a normal repo.

Reply:
> The reason for the separate repository is that the gensync-macports repository contains a different CMakeLists.txt than our main gensync repository.

Maintaining a separate codebase of your project just for MacPorts still seems like an utterly bizarre development decision to me. I don't understand why you want to bifurcate your user base in this way. Imagine we find a bug in gensync. Which of the two repos do we file the bug in? What if we want to search for an existing bug to see if a problem has already been reported? Now there are two issue trackers in which we need to search. What if we want to submit a fix? Which repo do we file the pull request in? How do you keep the code in the two repos synchronized? Why do you want to create all this additional work for yourselves?

The first sentence of the readme in that repo is:

> GenSync-macports is the extended version for MACOS.

So then it is not only for MacPorts? It is also for non-MacPorts users on macOS?

Looking at this CMakeLists.txt, I see that this software [requires C++11](https://github.com/nislab/gensync-macports/blob/0897f0f18765068fd8c54238de77fbc426b3051f/CMakeLists.txt#L3). This must be reflected in the Portfile via the line `compiler.cxx_standard 2011`.